### ISSUE TITLE
Use clone of MPI_COMM_WORLD in DART to avoid interference with application

### ISSUE THREAD

**Issue by fmoessbauer** [dash-project/dash#209]
For libraries using MPI it is good practice to duplicate the `MPI_COMM_WORLD` for inner-library communication to avoid interference with the MPI calls of the application.

This is especially necessary if using threads and `MPI_THREAD_MULTIPLE`. Otherwise library-internal barriers (as well as two-sided communications) might bind to application barriers.

**Example**
```cpp
std::async(std::launch::async, dash::accumulate(...));
// this barrier will catch the barrier inside dash::accumulate
MPI_Barrier(MPI_COMM_WORLD);
```

**Comment by devreal** [comment]
Good point, I will take of it.

**Comment by fmoessbauer** [comment]
Implemented in PR #210
